CMakeLists.txt

```diff
@@ -150,14 +150,17 @@ if (INSTALL_HEADER_ONLY)
   check_cxx_compiler_flag("-fsycl" is_dpcpp)
   if(is_dpcpp)
     target_compile_definitions(portblas INTERFACE "SB_ENABLE_USM")
-    if ((${CMAKE_CXX_COMPILER_ID} STREQUAL "IntelLLVM"
-           AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 2024.1)
-         OR NOT ${CMAKE_CXX_COMPILER_ID} STREQUAL "IntelLLVM")
-       # Apply only for oneAPI releases >= 2024.1 OR for intel/llvm.
-       target_link_options(portblas INTERFACE "-mllvm=-enable-global-offset=false")
-       target_compile_options(portblas INTERFACE "-mllvm=-enable-global-offset=false")
-       message(STATUS "Adding -mllvm=-enable-global-offset=false to portblas")
-     endif()
+    check_cxx_compiler_flag("-mllvm=-enable-global-offset=false" support_disable_global_offset)
+    if (${support_disable_global_offset})
+      if ((${CMAKE_CXX_COMPILER_ID} STREQUAL "IntelLLVM"
+             AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 2024.1)
+           OR NOT ${CMAKE_CXX_COMPILER_ID} STREQUAL "IntelLLVM")
+         # Apply only for oneAPI releases >= 2024.1 OR for intel/llvm.
+         target_link_options(portblas INTERFACE "-mllvm=-enable-global-offset=false")
+         target_compile_options(portblas INTERFACE "-mllvm=-enable-global-offset=false")
+         message(STATUS "Adding -mllvm=-enable-global-offset=false to portblas")
+       endif()
+    endif()
```

Comment:
Why don't we need to check cuda / amd here?

Reply:
Thanks for your review, @hjabird.

The first [commit](https://github.com/codeplaysoftware/portBLAS/pull/519/commits/61aa2498e17d7f6858fe45bdc73aa591b2750839) implemented a check based upon the `TUNING_TARGET`. Unfortunately, it is not possible to implement a check over the SYCL triplet in case of header only compilation as suggested.

After an internal discussion, we decided to check the required flag was supported in order to apply it regardless of the `TUNING_TARGET` for allowing it also in case of non-tuned configurations.